libcontainer/cgroups/fs/memory.go

```diff
@@ -147,6 +141,40 @@ func memoryAssigned(cgroup *configs.Cgroup) bool {
 		cgroup.MemorySwappiness != -1
 }
 
+func getMemoryMappingData(path, name string) (map[string]uint64, error) {
+	data := make(map[string]uint64)
+
+	file, err := os.Open(filepath.Join(path, name))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	defer file.Close()
+
+	sc := bufio.NewScanner(file)
+	for sc.Scan() {
+		t, v, err := getCgroupParamKeyValue(sc.Text())
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse %s (%q) - %v", name, sc.Text(), err)
+		}
+		data[t] = v
+	}
+
+	return data, nil
+}
+
+func getMemoryStat(path string) (map[string]uint64, error) {
+	// Get stats from memory.stat.
+	return getMemoryMappingData(path, "memory.stat")
+}
+
+func getOomStat(path string) (map[string]uint64, error) {
```

Comment:
Yeah I understand, since it's a yes or no, not a number, it looks more like `Status`, but it's not the same thing as the states we are implementing in https://github.com/opencontainers/runc/pull/311

So we end up facing states, status, statistic and events. I don't feel good to mix events and statistic into `runc events`, and it seems to be worse to add status in. Do you think we should separate them all? That would be a lot more commands but seems unavoidable.